tutorials/Inference-Bert-Model-for-High-Performance-with-ONNX-Runtime-on-AzureML.ipynb

```diff
@@ -0,0 +1,690 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3.2 - Write scoring file\n",
+    "\n",
+    "We are now going to deploy our ONNX model on Azure ML using the ONNX Runtime. We begin by writing a score.py file that will be invoked by the web service call. The `init()` function is called once when the container is started so we load the model using the ONNX Runtime into a global session object. Then the `run()` function is called when we run the model using the Azure ML web service. Add neccessary `preprocess()` and `postprocess()` steps. The following score.py file uses `bert-squad` as an example and assumes the inputs will be in the following format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs_json = {\n",
+    "  \"version\": \"1.4\",\n",
+    "  \"data\": [\n",
+    "    {\n",
+    "      \"paragraphs\": [\n",
+    "        {\n",
+    "          \"context\": \"In its early years, the new convention center failed to meet attendance and revenue expectations.[12] By 2002, many Silicon Valley businesses were choosing the much larger Moscone Center in San Francisco over the San Jose Convention Center due to the latter's limited space. A ballot measure to finance an expansion via a hotel tax failed to reach the required two-thirds majority to pass. In June 2005, Team San Jose built the South Hall, a $6.77 million, blue and white tent, adding 80,000 square feet (7,400 m2) of exhibit space\",\n",
+    "          \"qas\": [\n",
+    "            {\n",
+    "              \"question\": \"where is the businesses choosing to go?\",\n",
+    "              \"id\": \"1\"\n",
+    "            },\n",
+    "            {\n",
+    "              \"question\": \"how may votes did the ballot measure need?\",\n",
+    "              \"id\": \"2\"\n",
+    "            },\n",
+    "            {\n",
+    "              \"question\": \"When did businesses choose Moscone Center?\",\n",
+    "              \"id\": \"3\"\n",
+    "            }\n",
+    "          ]\n",
+    "        }\n",
+    "      ],\n",
+    "      \"title\": \"Conference Center\"\n",
+    "    }\n",
+    "  ]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile score.py\n",
+    "import os\n",
+    "import collections\n",
+    "import json\n",
+    "import time\n",
+    "from azureml.core.model import Model\n",
+    "import numpy as np    # we're going to use numpy to process input and output data\n",
+    "import onnxruntime    # to inference ONNX models, we use the ONNX Runtime\n",
+    "import wget\n",
+    "from pytorch_pretrained_bert.tokenization import whitespace_tokenize, BasicTokenizer, BertTokenizer\n",
+    "\n",
+    "def init():\n",
+    "    global session, tokenizer\n",
+    "    model = Model.get_model_path(model_name = 'bert-squad-large-uncased')\n",
```

Comment:
i would recommend using env var approach


Deployment | Environment variable value
-- | --
Single model | The path to the folder containing the model.
Multiple models | The path to the folder containing all models. Models are located by name and version in this folder ($MODEL_NAME/$VERSION)

Reply:
model_path = os.path.join(os.getenv('AZUREML_MODEL_DIR'), 'sklearn_regression_model.pkl')